code/__HELPERS/_macros.dm

```diff
@@ -137,6 +137,8 @@
 
 #define iscamera(A) istype(A, /obj/machinery/camera)
 
+#define isfreezer(A) is_type_in_list(A, list(/obj/structure/closet/secure_closet/freezer, /obj/structure/closet/crate/freezer/surgery, /obj/machinery/smartfridge))
```

Comment:
`is_type_in_list()` works best on lists that are stored somewhere rather than just created in each call, due to how its caching works.
I'd recommend either making a global list of freezers, or a static list on food items

Reply:
Sure